### PR TITLE
code homogenization

### PR DESCRIPTION
**Pull request by rdubigny**
- homogenize data and oauth controller
- generalize usage of validator to data controller
- fix typos
- set dataAuthorize eIDAS level to 1
- some variable renaming
- remove erroneous comments

### CODE REVIEW

File: controllers/oauthData.js [hunk 1]
```diff
@@ -7,26 +7,18 @@ import config from '../config';
  * to the France Connect Authorization and logout API endpoint.
  * @see @link{ https://partenaires.franceconnect.gouv.fr/fcp/fournisseur-service# }
  */
-export const oauthDataAuthorize = (req, res) => (
-  res.redirect(
+export const oauthDataAuthorize = (req, res) => {
+  const eidasQueryString = '&acr_values=eidas1';
+
```
Comment (Arnaud-Dev-Nodejs) [inline controllers/oauthData.js]:
```
export const oauthDataAuthorize = (req, res) => {
  const eidasQueryString = '&acr_values=eidas1';
  const allScopes = encodeURIComponent(`${config.MANDATORY_SCOPES} ${config.DGFIP_SCOPES}`); 
  return res.redirect(
    `${config.FC_URL}${config.AUTHORIZATION_FC_PATH}?`
      + `response_type=code&client_id=${config.DATA_CLIENT_ID}&redirect_uri=${config.FS_URL}`
      + `${config.DATA_CALLBACK_FS_PATH}&scope=${allScopes}&state=home&nonce=customNonce11`
      + `${eidasQueryString}`,
  );
};
```

File: helpers/utils.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-export const getAcrFromIdToken = idToken => (
-  JSON.parse(Buffer.from(idToken.split('.')[1], 'base64').toString('utf8')).acr
+export const getPayloadOfIdToken = idToken => (
+  JSON.parse(Buffer.from(idToken.split('.')[1], 'base64').toString('utf8'))
```
Comment (Arnaud-Dev-Nodejs) [inline helpers/utils.js]:
```
const [,payload = ""] = idToken.split('.');
const data = Buffer.from(payload, 'base64').toString('utf8');
try {
return JSON.parse(data);
} catch(JSONError) {
  console.error(JSONError);
 throw new Error('Invalid token parsing');
}

```

Comment (rdubigny) [reply helpers/utils.js]: https://stackoverflow.com/a/38552302/2590861

Comment (Arnaud-Dev-Nodejs) [reply helpers/utils.js]: console.error(JSONError, 'data:', data);

File: controllers/oauthData.js (hunk 1 above)
Comment (Arnaud-Dev-Nodejs) [reply controllers/oauthData.js]: const SEP = ' ';
const allScopes =  encodeURIComponent(
  [...config.MANDATORY_SCOPES.split(SEP), ...config.DGFIP_SCOPES.split(SEP)].join(SEP)
);

**Pull request merged**